code/mob/living/critter/ai/mimic.dm

```diff
@@ -0,0 +1,155 @@
+	transition_tasks += holder.get_instance(/datum/aiTask/timed/wait_in_ambush, list(holder, src))
+	transition_tasks += holder.get_instance(/datum/aiTask/sequence/goalbased/run_and_hide, list(holder, src))
+	//wander and attack are only called if nothing else is possible
+	transition_tasks += holder.get_instance(/datum/aiTask/timed/wander/critter/aggressive, list(holder, src))
+	transition_tasks += holder.get_instance(/datum/aiTask/sequence/goalbased/critter/attack, list(holder, src))
+
+/// Wait in ambush - while hidden, stay hidden until a potential target is nearby, then interupt AI so ambush or attack can trigger
+/datum/aiTask/timed/wait_in_ambush
+	name = "waiting in ambush"
+	minimum_task_ticks = 5
+	maximum_task_ticks = 10
+	weight = 12 //Higher than attack - if we can wait in ambush instead of attacking, we should.
+
+/datum/aiTask/timed/wait_in_ambush/evaluate()
+	var/mob/living/critter/mimic/C = holder.owner
+	if(length(C.seek_target(3)))
+		return 0
+	return 100*C.is_hiding*weight  //Note goalbased evaluate returns a percentage which is multiplied by a weight. Only return a high priority if we're hiding, otherwise 0
+
+/datum/aiTask/timed/wait_in_ambush/on_tick()
+	var/mob/living/critter/mimic/C = holder.owner
+	holder.stop_move()
+	if(length(C.seek_target(3))) //ambush max dist
+		src.holder.owner.ai.interrupt()
+
+/// If a target is in range and we're hiding, attack them until they're incapacitated
+/datum/aiTask/sequence/goalbased/ambush
+	name = "ambushing"
+	weight = 13 //higher than waiting in ambush
+	max_dist = 3
+	ai_turbo = TRUE
+
+/datum/aiTask/sequence/goalbased/ambush/New(parentHolder, transTask)
+	..()
+	add_task(holder.get_instance(/datum/aiTask/succeedable/ambush, list(holder)))
+
+/datum/aiTask/sequence/goalbased/ambush/precondition()
+	var/mob/living/critter/mimic/C = holder.owner
+	return C.is_hiding && C.can_critter_attack()
+
+/datum/aiTask/sequence/goalbased/ambush/get_targets()
+	var/mob/living/critter/C = holder.owner
+	return C.seek_target(src.max_dist)
+
+/datum/aiTask/sequence/goalbased/ambush/on_tick()
+	var/mob/living/critter/mimic/C = holder.owner
+	if(C.is_hiding)
+		C.stop_hiding()
+	. = ..()
+
+/datum/aiTask/succeedable/ambush
+	name = "ambush subtask"
+	max_dist = 5
+	var/has_started = FALSE
+
+/datum/aiTask/succeedable/ambush/failed()
+	//failure condition is just that the target escaped
+	if(holder.owner && holder.target)
+		return (GET_DIST(holder.owner, holder.target) > max_dist)
+	else
+		return TRUE //we also fail if owner or target are somehow null
+
+/datum/aiTask/succeedable/ambush/succeeded()
+	if(is_incapacitated(holder.target)) //attack until target is incapacitated
+		return TRUE
+	return FALSE
+
+/datum/aiTask/succeedable/ambush/on_tick()
+	//keep moving towards the target and attacking them in range for as long as is necessary
+	//has_started marks that we've hit them once
+	var/mob/living/critter/C = holder.owner
+	var/mob/T = holder.target
+	if(C && T && BOUNDS_DIST(C, T) == 0)
```

Comment:
non standard single letter variables!!1

Reply:
**C**ritter, **T**arget

Reply:
C is understandable, T is definitely not.